packages/forms-web-app/src/dynamic-forms/validator/validator.js

```diff
@@ -1,49 +1,38 @@
-// const { body, validationResult } = require('express-validator');
-// const hasQuestionnaire = require('../definitions/questionnaire/hasQuestionnaire');
-// const questionnaire = hasQuestionnaire.questionnaire;
-// const formHelper = require('../lib/dynamicformhelper');
+const RequiredValidator = require('./requiredValidator');
+const ValidOptionValidator = require('./validOptionValidator');
+const hasJourney = require('../has-questionnaire/journey');
 
-// const validate = () => {
-//     return async (req, res, next) => {
-//         const { appealId, section, question } = req.params;
-//         var questionObj = formHelper.getQuestionBySectionAndName(questionnaire, section, question);
-//         const validations = [];
-//         switch (questionObj.validator?.type) {
-//             case "text":
-//                 validations.push(textValidations(questionObj))
-//                 break;
-//             case "boolean":
-//                 validations.push(booleanValidations(questionObj))
-//                 break;
-//             case "boolean-text":
-//                 validations.push(booleanValidations(questionObj))
-//                 validations.push(textValidations(questionObj))
-//                 break;
-//         }
-//         await Promise.all(validations.map(validation => validation.run(req)));
+const validate = () => {
+	return async (req, res, next) => {
+		const { section, question } = req.params;
+		var questionObj = hasJourney.getQuestionBySectionAndName(section, question);
+		const validations = [];
+		questionObj.validators.forEach((validator) => {
+			switch (validator.constructor) {
+				case RequiredValidator:
+					validations.push(new RequiredValidator().validate(questionObj));
+					break;
+				case ValidOptionValidator:
+					validations.push(new ValidOptionValidator().validate(questionObj));
+					break;
+				default:
+					throw new Error(`Validator of type ${validator.constructor} not implemented`);
+			}
+		});
 
-//        next();
-//     };
-// };
+		const results = [];
+		let overallResult = true;
+		for (const validation of validations) {
+			let validationResult = await validation.run(req);
+			results.push(validationResult);
+			if (validationResult.errors.length > 0) {
+				next();
+				overallResult = false;
```

Comment:
Not sure these will get called as we're calling next() on the middleware pipeline

Reply:
It's not a javascript thing but rather part of the express middleware, because we're registering the middleware on the route we're ceding control to the express middleware pipeline so in this context calling next() will tell the express middleware to stop executing and hand over to the next middleware - e.g we've raise an error, stop processing and call the next middleware

Reply:
Ah I see because we're not explicitly calling return next() it will execute the rest of the stack. Do we want to just call return next() then to stop execution and move onto the error handler middleware as soon as an error is raised then?